pipelines/google_sheets/google_sheets.py

```diff
@@ -0,0 +1,133 @@
+# Contains the main pipeline functions
+
+from typing import Any, Iterator, cast
+import dlt
+from dlt.common.configuration.specs import GcpClientCredentialsWithDefault
+from dlt.common.typing import DictStrAny
+from dlt.common.exceptions import MissingDependencyException
+from dlt.common.typing import TDataItems
+import logging
+import time
+
+try:
+    from data_validator import serial_date_to_datetime, process_url
+except ModuleNotFoundError:
+    from google_sheets.data_validator import process_url, serial_date_to_datetime
+
+try:
+    from apiclient.discovery import build, Resource
+except ImportError:
+    raise MissingDependencyException("Google Sheets Source", ["google1", "google2"])
+
+# constants
+SPREADSHEET_ID = "1NVxFQYRqrGmur_MeIc4m4ewToF802uy2ObC61HOCstU"
```

Comment:
this is test code I assume?

Reply:
Yes, it will be cleaned up and more appropriate tests will be set up.